lib/Core/Executor.cpp

```diff
@@ -1544,36 +1559,48 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
 #else
           BasicBlock *caseSuccessor = si->getSuccessor(i);
 #endif
-          std::map<BasicBlock*, ref<Expr> >::iterator it =
-            targets.insert(std::make_pair(caseSuccessor,
-                           ConstantExpr::alloc(0, Expr::Bool))).first;
 
-          it->second = OrExpr::create(match, it->second);
+          // TODO add conditional
+          std::pair<std::map<BasicBlock *, ref<Expr> >::iterator, bool> res =
+              targets.insert(std::make_pair(
+                  caseSuccessor, ConstantExpr::alloc(0, Expr::Bool)));
+
+          res.first->second = OrExpr::create(match, res.first->second);
```

Comment:
@MartinNowack Ah I see. That makes sense now. This is worthy of a comment I think!